src/main/kotlin/com/trackforever/controllers/ProjectController.kt

```diff
@@ -96,58 +103,132 @@ class ProjectController (@Autowired private var projectRepository: ProjectReposi
     }
 
     /**
-     * Adds or updates a project based on the the given data.
+     * Adds or updates issues for projects.
      * Expects a mapping of K-V pairs such that K = projectKey and V = List of Issues as a JSON String
-     * Possible Responses: HTTP 200 OK if all projects exist
-     *                     HTTP 207 MULTI_STATUS with projectKeys mapped to an List of issues that failed to be set.
+     * Possible Responses: HTTP 200 OK with a map of all requested projects to their list of issues
+     *                     HTTP 207 MULTI_STATUS with all projects that could be found mapped to their list of issues
+     *                     HTTP 410 GONE if none of the projects could be found.
      */
     @CrossOrigin
     @PutMapping("/issues")
     fun setIssues(@RequestBody issues: String): ResponseEntity<Map<ProjectId, List<TrackForeverIssue>>> {
```

Comment:
This should be Map<ProjectId, Map<IssueId, Hash>>

Reply:
The Hash should be the new calculated hash for the updated issue

Reply:
Are we assuming that IssueIds can never change?

Reply:
Yes, we are. There will be an issue with issueid (with multiple clients) clashes, but that's not something we have time to fix.